packages/postgres/src/pool.ts

```diff
@@ -109,6 +109,10 @@ function paramsToArray(
   }
   processedSql += remainingSql;
   remainingSql = '';
+  const difference = _.difference(Object.keys(params), Object.keys(map));
+  if (difference.length) {
```

Comment:
`console.warn` is unacceptable for production code. Use `logger.warn` if you must, but I much prefer the idea on Slack to throw an error when not in production. No one will *ever* read warning logs where they matter; not in CI, and certainly not in production.

Reply:
Oh I fully understand. I just put this out here as a conversation starter. How do you suggest the package knows about running in production vs test/development?

Reply:
I don't think it should! Rather, let's add an option that explicitly configures this behavior, e.g. add a `setErrorOnUnusedParameter(enabled: boolean)` function to the pool, then call that with an appropriate value, e.g. `config.devMode` in the app and `true` in tests. Alternatively, add this to the options object that `Pool.initAsync` accepts.